src/services/Redis.ts

```diff
@@ -1,63 +1,110 @@
 import { ClientOpts, createClient, RedisClient } from 'redis';
-import { promisify } from 'util';
+import { EventEmitter } from 'events';
 import Expire from '../types/Expire';
 import Service from '../types/Service';
 
+enum methodLabels {
+    get = 'get',
+    set = 'set',
+    del = 'del',
+}
+
 class Redis implements Service {
-    public commands: {
-        del: (key: string) => Promise<void>;
-        expire: (key: string, seconds: number) => Promise<void>;
-        expireAt: (key: string, seconds: number) => Promise<void>;
-        get: (key: string) => Promise<string>;
-        set: (key: string, value: string) => Promise<void>;
-    };
+    /** @type RedisClient */
+    private client: RedisClient = null;
+
+    /** @type ClientOpts */
+    private config: ClientOpts;
 
-    private client: RedisClient;
+    /** @type Boolean */
+    private ephemeral: boolean;
+
+    /** @type Boolean */
+    private listenersRegistered: boolean;
+
+    /** @type EventEmitter */
+    private callbacks: EventEmitter = new EventEmitter();
 
     /**
-     * Creates a redis client instance.
+     * Creates a Redis client instance depending on config.
      */
-    public constructor(config?: ClientOpts) {
-        this.client = createClient(config);
-        this.commands = {
-            del: promisify(this.client.del).bind(this.client),
-            expire: promisify(this.client.expire).bind(this.client),
-            expireAt: promisify(this.client.expireat).bind(this.client),
-            get: promisify(this.client.get).bind(this.client),
-            set: promisify(this.client.set).bind(this.client),
-        };
+    public constructor(config?: ClientOpts, ephemeral?: boolean) {
+        this.ephemeral = ephemeral;
+        this.config = config;
+        this.listenersRegistered = false;
```

Comment:
are we using it?

Reply:
I introduced listenersRegistered so that we don't re-register the listeners every time we make a redis call.

I noticed it was doing it when I was debugging on staging and first get I got one console.log, second get I got two, etc.

You can see in registerListeners() and getClient() what I do with it

Reply:
cool